pulp_container/app/registry_api.py

```diff
@@ -970,3 +970,42 @@ def receive_artifact(self, chunk):
                 artifact = Artifact.objects.get(sha256=artifact.sha256)
                 artifact.touch()
             return artifact
+
+
+class Signatures(ContainerRegistryApiMixin, ViewSet):
+    """A ViewSet for image signatures."""
+
+    lookup_value_regex = "sha256:[0-9a-f]{64}"
+
+    def head(self, request, path, pk=None):
+        """Respond to HEAD requests querying signatures by sha256."""
+        return self.get(request, path, pk=pk)
+
+    def get(self, request, path, pk):
+        """Return a signature identified by its sha256 checksum."""
+        _, _, repository_version = self.get_drv_pull(path)
+
+        try:
+            manifest = models.Manifest.objects.get(digest=pk)
+        except models.Manifest.DoesNotExit:
+            raise ManifestNotFound(reference=pk)
+
+        signatures = models.ManifestSignature.objects.filter(
+            signed_manifest=manifest, pk__in=repository_version.content
+        )
+
+        return Response(self.get_response_data(signatures))
+
+    @staticmethod
+    def get_response_data(signatures):
+        """Extract version, type, name, and content from the passed signature data."""
+        data = []
+        for signature in signatures:
+            signature = {
+                "schemaVersion": 2,
```

Comment:
that's not correct, version was correct https://github.com/containers/image/blob/b55fb86c28b7d743cf59701332cd78d4294c7c54/docker/docker_image_src.go#L559
https://github.com/containers/image/blob/f2c99a9b1a0cd6673aab0d538b48d68562702b04/docker/docker_client.go#L76

Reply:
Good find, "schemaVersion" worked much better!

@ipanova , https://github.com/containers/image/blob/f2c99a9b1a0cd6673aab0d538b48d68562702b04/docker/docker_client.go#L76 shows in the comment how json is being serialized, and that it's this sting and not the attribute name is used. But it is confusing, why not to make them the same :woman_shrugging:

Reply:
ok that's confusing as hell

Reply:
`Version` stands for a parsed value. A raw JSON value is stored as "schemaVersion".